### ISSUE TITLE
Share component is not formating the email address correctly and giving an error

### ISSUE THREAD

**Issue by mgurney-leappoint** [adobe/asset-share-commons#908]
Hi,

I am new to AEM and Asset Share Commons, we have a site running and are using the share component, this has been working until recently, now it is throwing the folllowing error in the logs and it appears that the email address is not formatted correctly.

```
[1] WARN [com.adobe.aem.commons.assetshare.util.impl.EmailServiceImpl] Invalid email address mgurney&amp;#64;leappoint.com passed to sendEmail(). Skipping.
```

Malformed email: `mgurney&amp;#64;leappoint.com`

Any ideas why the html encoding is not getting replaced with the `@` symbol ?

Regards
Michael

**Comment by davidjgonzalez** [comment]
Let me take a look - it might be overly aggressive xss encoding if input parameters

**Comment by davidjgonzalez** [comment]
[assetshare.core-2.6.0-SNAPSHOT.renametojar.txt](https://github.com/adobe/asset-share-commons/files/10855721/assetshare.core-2.6.0-SNAPSHOT.renametojar.txt)

@mgurney-leappoint any possibility you can test this bundle out and see if it fixes the issue?

**Comment by mgurney-leappoint** [comment]
@davidjgonzalez Sorry for my basic knowledge but how would i install this jar file, can i deploy this through the repository ? This is a clients AEM as a Cloud so don't have access to web console to upload and start the bundle.

**Comment by davidjgonzalez** [comment]
@mgurney-leappoint mmm - well that doesnt make it easy :)

Were you able to reproduce this issue on AEM SDK? If so you can install it directly there.
Or if you have a AEM CS RDE env, you could push it directly to that.

**Comment by mgurney-leappoint** [comment]
Hi David,

Unfortunately we don’t have an RDE environment for me to test this in, I will try and reproduce this locally on my SDK to see if this resolves the issue.

Michael Gurney
Solution Principal - EMEA
E.
***@***.******@***.***>
W.
leappoint.com<https://leappoint.com/>
[signature_1655988605]<https://leappoint.com/>
[signature_4167622414]<https://www.linkedin.com/leappoint>
[signature_279844203]<https://www.twitter.com/leappointllc>
[signature_3850742834]<https://www.facebook.com/LeapPoint-155162091221401>
[signature_2499086105]<https://www.instagram.com/leappoint>
[signature_2138334130]<https://www.youtube.com/channel/UCz_rJxXkbTiF49yGVeKbktA/featured>
The content of this email is confidential and intended for the recipient specified in message only. It is strictly forbidden to share any part of this message with any third party, without a written consent of the sender. If you received this message by mistake, please reply to this message and follow with its deletion, so that we can ensure such a mistake does not occur in the future.


From: david g. ***@***.***>
Date: Wednesday, 1 March 2023 at 14:40
To: adobe/asset-share-commons ***@***.***>
Cc: Michael Gurney ***@***.***>, Mention ***@***.***>
Subject: Re: [adobe/asset-share-commons] Share component is not formating the email address correctly and giving an error (Issue #908)

@mgurney-leappoint<https://github.com/mgurney-leappoint> mmm - well that doesnt make it easy :)

Were you able to reproduce this issue on AEM SDK? If so you can install it directly there.
Or if you have a AEM CS RDE env, you could push it directly to that.

**Comment by davidjgonzalez** [comment]
Took a 2nd look at the mailformed email... this is very odd:

Malformed email: mgurney&amp;#64;leappoint.com

What this ~ translates to is:

+ mgurney
+ `&amp;` -> `&`
+ `#64;` -> ASCII for `@`
+ leappoint.com

@mgurney-leappoint What is the email address(es) you are actually entering the form? Any idea where that `&` would be coming from?

**Comment by davidjgonzalez** [comment]
@mgurney-leappoint - Can you verify what version of AEM as a Cloud Service and Asset Share Commons you are using?

Can you re-produce this on your AEM SDK? I cannot reproduce this.

**Comment by mgurney-leappoint** [comment]
Hi David,


Its actually wrong in the ticket as it keeps encoding the HTML that I enter, it should be mgurney&amp;#64;leappoint.com



Invalid email address mgurney&amp;#64;leappoint.com passed to sendEmail(). Skipping.

My email is ***@***.***

Michael Gurney
Solution Principal - EMEA
E.
***@***.******@***.***>
W.
leappoint.com<https://leappoint.com/>
[signature_4248060312]<https://leappoint.com/>
[signature_2797031837]<https://www.linkedin.com/leappoint>
[signature_1209450875]<https://www.twitter.com/leappointllc>
[signature_2940204899]<https://www.facebook.com/LeapPoint-155162091221401>
[signature_408568879]<https://www.instagram.com/leappoint>
[signature_4061268037]<https://www.youtube.com/channel/UCz_rJxXkbTiF49yGVeKbktA/featured>
The content of this email is confidential and intended for the recipient specified in message only. It is strictly forbidden to share any part of this message with any third party, without a written consent of the sender. If you received this message by mistake, please reply to this message and follow with its deletion, so that we can ensure such a mistake does not occur in the future.


From: david g. ***@***.***>
Date: Thursday, 2 March 2023 at 15:42
To: adobe/asset-share-commons ***@***.***>
Cc: Michael Gurney ***@***.***>, Mention ***@***.***>
Subject: Re: [adobe/asset-share-commons] Share component is not formating the email address correctly and giving an error (Issue #908)

Took a 2nd look at the mailformed email... this is very odd:

Malformed email: mgurney&#64<https://github.com/adobe/asset-share-commons/pull/64>;leappoint.com

What this ~ translates to is:

  *   mgurney
  *   &amp; -> &
  *   #64; -> ASCII for @
  *   leappoint.com

@mgurney-leappoint<https://github.com/mgurney-leappoint> What is the email address(es) you are actually entering the form? Any idea where that & would be coming from?

**Comment by davidjgonzalez** [comment]
@mgurney-leappoint it seems youre on ASC v2.3.2 - which is very old - June 2022!!!

This was fixed somewhere along the way since then (dont recall the exact version) but its resolved in latest version (v2.5.4) -- please make sure you're using the latest version of ASC.

**Comment by davidjgonzalez** [comment]
For posterity, im not sure if this is what fixed it between 2.3.2 and 2.5.4: https://github.com/adobe/asset-share-commons/pull/887 ... its the only thing i can see that makes any sense. 

Email sending has worked for years, well before https://github.com/adobe/asset-share-commons/pull/887 ... so its quite strange.

**Comment by mgurney-leappoint** [comment]
Hi David,

This is fixed now that I have upgraded to the latest version 2.5.4.

Thank you for your help.

Michael Gurney
Solution Principal - EMEA
E.
***@***.******@***.***>
W.
leappoint.com<https://leappoint.com/>
[signature_1874167819]<https://leappoint.com/>
[signature_3696798502]<https://www.linkedin.com/leappoint>
[signature_2569195861]<https://www.twitter.com/leappointllc>
[signature_1664542983]<https://www.facebook.com/LeapPoint-155162091221401>
[signature_181447663]<https://www.instagram.com/leappoint>
[signature_70161210]<https://www.youtube.com/channel/UCz_rJxXkbTiF49yGVeKbktA/featured>
The content of this email is confidential and intended for the recipient specified in message only. It is strictly forbidden to share any part of this message with any third party, without a written consent of the sender. If you received this message by mistake, please reply to this message and follow with its deletion, so that we can ensure such a mistake does not occur in the future.


From: david g. ***@***.***>
Date: Friday, 3 March 2023 at 19:06
To: adobe/asset-share-commons ***@***.***>
Cc: Michael Gurney ***@***.***>, Mention ***@***.***>
Subject: Re: [adobe/asset-share-commons] Share component is not formating the email address correctly and giving an error (Issue #908)

For posterity, im not sure if this is what fixed it between 2.3.2 and 2.5.4: #887<https://github.com/adobe/asset-share-commons/pull/887> ... its the only thing i can see that makes any sense.

Email sending has worked for years, well before #887<https://github.com/adobe/asset-share-commons/pull/887> ... so its quite strange.

**Comment by davidjgonzalez** [comment]
Great to hear!
